platforms/jvm/scala/src/main/java/org/gradle/api/internal/tasks/scala/MinimalScalaCompileOptions.java

```diff
@@ -42,17 +42,17 @@ public class MinimalScalaCompileOptions implements Serializable {
     private final KeepAliveMode keepAliveMode;
 
     public MinimalScalaCompileOptions(BaseScalaCompileOptions compileOptions) {
-        this.failOnError = compileOptions.isFailOnError();
-        this.deprecation = compileOptions.isDeprecation();
-        this.unchecked = compileOptions.isUnchecked();
-        this.debugLevel = compileOptions.getDebugLevel();
-        this.optimize = compileOptions.isOptimize();
-        this.encoding = compileOptions.getEncoding();
-        this.force = compileOptions.isForce();
-        this.additionalParameters = ImmutableList.copyOf(compileOptions.getAdditionalParameters());
-        this.listFiles = compileOptions.isListFiles();
-        this.loggingLevel = compileOptions.getLoggingLevel();
-        this.loggingPhases = compileOptions.getLoggingPhases() == null ? null : ImmutableList.copyOf(compileOptions.getLoggingPhases());
+        this.failOnError = compileOptions.getFailOnError().getOrElse(true);
+        this.deprecation = compileOptions.getDeprecation().getOrElse(true);
+        this.unchecked = compileOptions.getUnchecked().getOrElse(true);
+        this.debugLevel = compileOptions.getDebugLevel().getOrNull();
+        this.optimize = compileOptions.getOptimize().getOrElse(false);
+        this.encoding = compileOptions.getEncoding().getOrNull();
+        this.force = compileOptions.getForce().getOrElse(false);
+        this.additionalParameters = ImmutableList.copyOf(compileOptions.getAdditionalParameters().get());
+        this.listFiles = compileOptions.getListFiles().getOrElse(false);
+        this.loggingLevel = compileOptions.getLoggingLevel().getOrNull();
+        this.loggingPhases = ImmutableList.copyOf(compileOptions.getLoggingPhases().get());
```

Comment:
❓ Why do we need to pass default values here? Don't we have conventions set for all of these?

I don't see how we can avoid introducing something like a `BooleanFlagProperty` to properly handle these use cases. We need a way to encode in the type system that these properties can never be absent.

Reply:
I will leave this for now. I can just remove defaults for inputs that are not `@Optional` (or `@Internal` or `@Console`) if you want. But I generally think it's not a big problem

Reply:
The problem I see here is that we are specifying `true` as a default for `failOnError` here, but we also do the same in `BasicScalaOptions`. It should not need to happen in both places. What if the values are different? (Essentially, the value here is useless, as `failOnError` will never be absent. But then why do we have this `true` here?)

I'm still convinced we need a way for the type system to express that these things cannot be absent.

Reply:
I think leaving this as is for now is okay, introducing a non-nullable property type should happen on its own.